bindinfo/handle.go

```diff
@@ -609,10 +610,104 @@ func (c cache) getBindRecord(hash, normdOrigSQL, db string) *BindRecord {
 	return nil
 }
 
+type captureFilter struct {
+	dbs       map[string]struct{}
+	frequency int64
+	tables    map[stmtctx.TableEntry]struct{}
+
+	fail      bool
+	currentDB string
+}
+
+func (cf *captureFilter) Enter(in ast.Node) (out ast.Node, skipChildren bool) {
+	switch x := in.(type) {
+	case *ast.TableName:
+		tblEntry := stmtctx.TableEntry{
+			DB:    x.Schema.L,
+			Table: x.Name.L,
+		}
+		if x.Schema.L == "" {
+			tblEntry.DB = cf.currentDB
+		}
+		if _, ok := cf.dbs[tblEntry.DB]; ok {
+			cf.fail = true
+		} else if _, ok := cf.tables[tblEntry]; ok {
+			cf.fail = true
+		}
+	}
+	return in, cf.fail
+}
+
+func (cf *captureFilter) Leave(in ast.Node) (out ast.Node, ok bool) {
+	return in, true
+}
+
+func (cf *captureFilter) isEmpty() bool {
+	return len(cf.dbs) == 0 && len(cf.tables) == 0
+}
+
+func (h *BindHandle) extractCaptureFilterFromStorage() (filter *captureFilter) {
+	filter = &captureFilter{
+		dbs:       make(map[string]struct{}),
+		frequency: 1,
+		tables:    make(map[stmtctx.TableEntry]struct{}),
+	}
+	exec := h.sctx.Context.(sqlexec.RestrictedSQLExecutor)
+	stmt, err := exec.ParseWithParams(context.TODO(), `SELECT filter_type, filter_value FROM mysql.capture_plan_baselines_blacklist order by filter_type`)
+	if err != nil {
+		logutil.BgLogger().Warn("[sql-bind] failed to parse query for mysql.capture_plan_baselines_blacklist load", zap.Error(err))
+		return
+	}
+	// No need to acquire the session context lock for ExecRestrictedStmt, it
+	// uses another background session.
+	rows, _, err := exec.ExecRestrictedStmt(context.TODO(), stmt)
+	if err != nil {
+		logutil.BgLogger().Warn("[sql-bind] failed to load mysql.capture_plan_baselines_blacklist", zap.Error(err))
+		return
+	}
+	for _, row := range rows {
+		filterTp := strings.ToLower(row.GetString(0))
+		valStr := strings.ToLower(row.GetString(1))
+		switch filterTp {
+		case "db":
+			filter.dbs[valStr] = struct{}{}
+		case "table":
+			strs := strings.Split(valStr, ".")
```

Comment:
Should we handle more situations like there are some spaces in the string?

Reply:
We can do it later if users really insert stupid records like that.